src/Controller/TagController.php

```diff
@@ -33,6 +34,9 @@ public function newTagAction(Request $request)
 
     public function tagAction($slug, Request $request)
     {
+        if (!$this->isGranted("ROLE_ADMIN")) {
+            throw new AccessDeniedException();
+        }
```

Comment:
Can be simplified to `$this->denyAccessUnlessGranted('ROLE_ADMIN');` (see [docs](https://symfony.com/doc/5.4/security/voters.html))